Language Switcher + Translations

satanmourner opened the pull request:
**CHORE/FEAT:**
- [x] added language switcher for selecting supported languages (Arabic, English, Spanish, French, Japanese, Korean, Chinese)
- [x] added translations for all pages

TODO:
- [ ] `pre-org-register` and `org-register` HTMLs
- [ ] check with Mohammad

mohammadhb reviewed (commented):
Perfect @satanmourner, Well done!!!

Here's what's working right now:
- Change language is working
- Changing language effect still persists when navigating to other pages
- RTL on selecting languages is working and updates the UI
- Initial language is being set

Small missing still might persist ( didn't check the design )
- Forget password page doesn't have drop down but translation is working perfectly, do we need one?

mohammadhb commented:
Fixed isssues:
1. the page `/auth/register/pre` does have the switcher and `chevron-down` is loading
2. default language would be set, if user doesn't have any preferances
3. Translation in  these pages are working:
- `/auth/register/pre`
- `/auth/register`
- `/organizations/register/pre`
- `/organizations/register`
- `/users/profile`
- `/auth/otp`
- `/auth/password/forget`

Known issues:
1. A 2-times refresh would reset the language to default[en] ( Couldn't test the pages more because of this issue )
2. In following pages, switcher doesn't exists:
- `/auth/register`
- `/organizations/register/pre`
- `/organizations/register`
- `/users/profile` 
4. Switcher would stay open even after selecting the language

satanmourner commented:
1. For me, on my local, everything works fine, but I can't test on the BE project. We can check it together later
2. We have no design, and I prefer to put it in `auth/login`, `auth/register/pre`, and `organization/register/pre` (please check the pre-register organization because I added it recently)
3. After selecting a language, Language Switcher will be closed (please check it too, because I fixed it)

@mohammadhb

mohammadhb commented:
> 1. For me, on my local, everything works fine, but I can't test on the BE project. We can check it together later
> 2. We have no design, and I prefer to put it in `auth/login`, `auth/register/pre`, and `organization/register/pre` (please check the pre-register organization because I added it recently)
> 3. After selecting a language, Language Switcher will be closed (please check it too, because I fixed it)
> 
> @mohammadhb

1. checked it and it was my browsers that were pushing the language
2. checked pre-register and its working properly
3. It's also fixed thanks

mohammadhb reviewed (approved):
Well done @satanmourner 💯
